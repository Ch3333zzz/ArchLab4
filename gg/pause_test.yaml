name: pause-test
config:
  pause_tick: 3
source:
  language: lisp
  code: |
    (print "A")
    (print "B")
    (print "C")
expect:
  out_stdout: A
  state: paused
  out_code: !!binary |
    AAAAFAAAAEYCAAAUAAAARgQAABQAAABGAAAAAQ==
  out_code_hex: |-
    0 - 00000014 - LOAD_CONST 0
    4 - 00000046 - PRINT
    8 - 02000014 - LOAD_CONST 2
    12 - 00000046 - PRINT
    16 - 04000014 - LOAD_CONST 4
    20 - 00000046 - PRINT
    24 - 00000001 - HALT
  out_log: |
    DEBUG root:processor.py:197 Datapath: heap_ptr initialized to word 1031
    DEBUG root:processor.py:616 No more scheduled input and mmio_in_buffer empty -> input closed (EOF)
    DEBUG root:processor.py:591 STATE: RUNNING    STEP: COMMAND_FETCH   TICK:    0 PC:  4096 ADDR:     0 MEM[ADDR]:          1 ACC:          0 SP: 64256 FSP: 64512 CP: 65536 C: 0 V: 0	INSTR: fetch
    DEBUG root:processor.py:591 STATE: RUNNING    STEP: OPERAND_FETCH   TICK:    0 PC:  4096 ADDR:     0 MEM[ADDR]:          1 ACC:          0 SP: 64256 FSP: 64512 CP: 65536 C: 0 V: 0	INSTR: LOAD_CONST 0
    DEBUG root:processor.py:694 Tick 0 PC 4096: LOAD_CONST 0 ACC=0 SP=64256 FSP=64512 CP=65536
    DEBUG root:processor.py:591 STATE: RUNNING    STEP: EXECUTION       TICK:    0 PC:  4100 ADDR:     1 MEM[ADDR]:         65 ACC:          0 SP: 64256 FSP: 64512 CP: 65536 C: 0 V: 0	INSTR: LOAD_CONST 0
    DEBUG root:processor.py:591 STATE: RUNNING    STEP: COMMAND_FETCH   TICK:    1 PC:  4100 ADDR:     1 MEM[ADDR]:         65 ACC:          0 SP: 64256 FSP: 64512 CP: 65536 C: 0 V: 0	INSTR: fetch
    DEBUG root:processor.py:591 STATE: RUNNING    STEP: OPERAND_FETCH   TICK:    1 PC:  4100 ADDR:     1 MEM[ADDR]:         65 ACC:          0 SP: 64256 FSP: 64512 CP: 65536 C: 0 V: 0	INSTR: PRINT
    DEBUG root:processor.py:694 Tick 1 PC 4100: PRINT 0 ACC=0 SP=64256 FSP=64512 CP=65536
    DEBUG root:processor.py:1086 PRINT: Detected pstr at 0 len=1
    DEBUG root:processor.py:320 [MMIO OUT] wrote 65 -> char: A
    DEBUG root:processor.py:591 STATE: RUNNING    STEP: EXECUTION       TICK:    1 PC:  4104 ADDR:     2 MEM[ADDR]:          1 ACC:          0 SP: 64256 FSP: 64512 CP: 65536 C: 0 V: 0	INSTR: PRINT
    DEBUG root:processor.py:591 STATE: RUNNING    STEP: COMMAND_FETCH   TICK:    2 PC:  4104 ADDR:     2 MEM[ADDR]:          1 ACC:          0 SP: 64256 FSP: 64512 CP: 65536 C: 0 V: 0	INSTR: fetch
    DEBUG root:processor.py:591 STATE: RUNNING    STEP: OPERAND_FETCH   TICK:    2 PC:  4104 ADDR:     2 MEM[ADDR]:          1 ACC:          0 SP: 64256 FSP: 64512 CP: 65536 C: 0 V: 0	INSTR: LOAD_CONST 2
    DEBUG root:processor.py:694 Tick 2 PC 4104: LOAD_CONST 2 ACC=0 SP=64256 FSP=64512 CP=65536
    DEBUG root:processor.py:591 STATE: RUNNING    STEP: EXECUTION       TICK:    2 PC:  4108 ADDR:     3 MEM[ADDR]:         66 ACC:          2 SP: 64256 FSP: 64512 CP: 65536 C: 0 V: 0	INSTR: LOAD_CONST 2
    DEBUG root:processor.py:591 STATE: PAUSED     STEP: PAUSE_CHECK     TICK:    3 PC:  4108 ADDR:     3 MEM[ADDR]:         66 ACC:          2 SP: 64256 FSP: 64512 CP: 65536 C: 0 V: 0	INSTR: pause
