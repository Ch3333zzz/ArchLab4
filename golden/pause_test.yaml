name: pause-test
config:
  pause_tick: 3
source:
  language: lisp
  code: |
    (print "A")
    (print "B")
    (print "C")
expect:
  out_stdout: A
  state: paused
  out_code: !!binary |
    FAAAAABGAAAAABQCAAAARgAAAAAUBAAAAEYAAAAAAQAAAAA=
  out_code_hex: |-
    0 - 1400000000 - LOAD_CONST 0
    5 - 4600000000 - PRINT
    10 - 1402000000 - LOAD_CONST 2
    15 - 4600000000 - PRINT
    20 - 1404000000 - LOAD_CONST 4
    25 - 4600000000 - PRINT
    30 - 0100000000 - HALT
  out_log: |
    DEBUG root:processor.py:357 No more scheduled input and mmio_in_buffer empty -> input closed (EOF)
    DEBUG root:processor.py:339 STATE: RUNNING    STEP: COMMAND_FETCH   TICK:    0 PC:  4096 ADDR:     0 MEM[ADDR]:          1 ACC:          0 C: 0 V: 0	INSTR: fetch
    DEBUG root:processor.py:339 STATE: RUNNING    STEP: OPERAND_FETCH   TICK:    0 PC:  4096 ADDR:     0 MEM[ADDR]:          1 ACC:          0 C: 0 V: 0	INSTR: LOAD_CONST 0
    DEBUG root:processor.py:396 Tick 0 PC 4096: LOAD_CONST 0 ACC=0
    DEBUG root:processor.py:339 STATE: RUNNING    STEP: EXECUTION       TICK:    0 PC:  4101 ADDR:     1 MEM[ADDR]:         65 ACC:          0 C: 0 V: 0	INSTR: LOAD_CONST 0
    DEBUG root:processor.py:339 STATE: RUNNING    STEP: COMMAND_FETCH   TICK:    1 PC:  4101 ADDR:     1 MEM[ADDR]:         65 ACC:          0 C: 0 V: 0	INSTR: fetch
    DEBUG root:processor.py:339 STATE: RUNNING    STEP: OPERAND_FETCH   TICK:    1 PC:  4101 ADDR:     1 MEM[ADDR]:         65 ACC:          0 C: 0 V: 0	INSTR: PRINT
    DEBUG root:processor.py:396 Tick 1 PC 4101: PRINT 0 ACC=0
    DEBUG root:processor.py:646 PRINT: Detected pstr at 0 len=1
    DEBUG root:processor.py:241 [MMIO OUT] wrote 65 -> char: A
    DEBUG root:processor.py:339 STATE: RUNNING    STEP: EXECUTION       TICK:    1 PC:  4106 ADDR:     2 MEM[ADDR]:          1 ACC:          0 C: 0 V: 0	INSTR: PRINT
    DEBUG root:processor.py:339 STATE: RUNNING    STEP: COMMAND_FETCH   TICK:    2 PC:  4106 ADDR:     2 MEM[ADDR]:          1 ACC:          0 C: 0 V: 0	INSTR: fetch
    DEBUG root:processor.py:339 STATE: RUNNING    STEP: OPERAND_FETCH   TICK:    2 PC:  4106 ADDR:     2 MEM[ADDR]:          1 ACC:          0 C: 0 V: 0	INSTR: LOAD_CONST 2
    DEBUG root:processor.py:396 Tick 2 PC 4106: LOAD_CONST 2 ACC=0
    DEBUG root:processor.py:339 STATE: RUNNING    STEP: EXECUTION       TICK:    2 PC:  4111 ADDR:     3 MEM[ADDR]:         66 ACC:          2 C: 0 V: 0	INSTR: LOAD_CONST 2
    DEBUG root:processor.py:339 STATE: PAUSED     STEP: PAUSE_CHECK     TICK:    3 PC:  4111 ADDR:     3 MEM[ADDR]:         66 ACC:          2 C: 0 V: 0	INSTR: pause
